Accept multiple prefixes

examosa opened the pull request:
Closes #12

Commit:
Use Array.prototype.reduce

Commit:
feature: accept multiple prefixes

Commit:
Remove failing script from travis.yml

bahmutov commented on .travis.yml:
```diff
@@ -9,8 +9,6 @@ node_js:
   - npm test
   - npm run travis-demo
-  # print names of all sensitive variables present
-  - ./bin/has-env.js NPM
```
try variable prefix that's for sure present

- ./bin/has-env.js TRAVIS

examosa replied:
Fixed

bahmutov commented:
I love it overall

bahmutov commented:
@Teomik129 I know this is probably a small thing, but would you like to be a collaborator on this repo? I am very short on time these days and it might be faster if you could review and merge your own PRs

bahmutov commented on src/print-env.spec.js:
```diff
@@ -0,0 +1,77 @@
+    const bin = join(__dirname, '..', 'bin', 'print-env.js')
+    it('prints sorted variables', async () => {
+      const { stdout } = await execa('node', [bin, 'FOOX'], {
+        env: process.env
```
we can use https://github.com/sindresorhus/execa#extendenv option to only pass `env` object and not the entire `env + process.env`

examosa replied:
Better yet, I just added to `process.env` and didn't pass any options to `execa`; see next commit.

bahmutov commented on src/print-env.spec.js:
```diff
@@ -0,0 +1,77 @@
+  let env
+
+  beforeEach(() => {
+    env = clone(process.env)
```
we can either use `extendEnv` (see below) or https://github.com/bahmutov/mocked-env to mock the environment object

examosa replied:
Node processes can't permanently alter the environment, so I believe we can safely overwrite `process.env` without having to restore it later.

examosa commented:
> @Teomik129 I know this is probably a small thing, but would you like to be a collaborator on this repo? I am very short on time these days and it might be faster if you could review and merge your own PRs

I would be more than happy to be a collaborator for this project!

bahmutov reviewed (dismissed):
need to explain new feature in README

Commit:
Updates

- Update README to include new features
- Move License
- Simplify spec
- Update .travis.yml

examosa reviewed (commented):
All good

Commit:
Merge branch 'master' into master
